File: services/configurator/src/core/logger.rs
use tracing_subscriber::{fmt, EnvFilter};

pub fn init_logging(log_level: &str) {
    fmt()
        .with_env_filter(EnvFilter::new(log_level))
        .init();
    
    tracing::info!("Logging initialized with level: {}", log_level);
}

File: 01-simple-rag/src/utils/logger.py
import logging
import os

from dotenv import load_dotenv

load_dotenv()

LOG_LEVEL = os.getenv("LOG_LEVEL", "INFO").upper()

LOG_FORMAT = "%(asctime)s | %(levelname)-8s | %(name)s | %(message)s"
LOG_DATE_FORMAT = "%Y-%m-%d %H:%M:%S"

_initialized = False


def get_logger(name: str) -> logging.Logger:
    global _initialized
    if not _initialized:
        logging.basicConfig(
            level=LOG_LEVEL,
            format=LOG_FORMAT,
            datefmt=LOG_DATE_FORMAT,
        )
        _initialized = True
    return logging.getLogger(name)